packages/http-client-java/generator/http-client-generator-test/src/test/java/tsptest/armstreamstyleserialization/StreamStyleSerializationTests.java

```diff
@@ -58,4 +66,42 @@ public void testValidate() {
         shark.withRequiredStringAnotherPropertiesRequiredString("any");
         shark.validate();
     }
+
+    @Test
+    public void testExpandableEnum() {
+        HttpClient httpClient = createExpandableEnumHttpClient();
+        // normal case
+        ArmStreamStyleSerializationManager manager = ArmStreamStyleSerializationManager
+            .authenticate(new HttpPipelineBuilder().httpClient(httpClient).build(), ArmUtils.getAzureProfile());
+        Priority priority = manager.priorities().setPriority(Priority.HIGH);
+        Assertions.assertEquals(Priority.HIGH, priority);
+
+        // null case
+        priority = manager.priorities().setPriority(Priority.HIGH);
+        Assertions.assertNull(priority);
+
+        // exception case, expected number, but received string
+        // azure-json wraps IllegalArgumentException
+        Assertions.assertThrows(RuntimeException.class, () -> manager.priorities().setPriority(Priority.HIGH));
+    }
+
+    private static HttpClient createExpandableEnumHttpClient() {
+        AtomicInteger callCount = new AtomicInteger();
+        HttpClient httpClient = request -> {
+            int count = callCount.incrementAndGet();
+            String query = request.getUrl().getQuery();
+            Assertions.assertEquals("priority=0", query);
+            if (count == 1) {
+                // normal case
+                return Mono.just(new MockHttpResponse(request, 200, 0));
+            } else if (count == 2) {
+                // null case
+                return Mono.just(new MockHttpResponse(request, 200, (Object) null));
+            } else {
+                // exception case, expected number, but received string
+                return Mono.just(new MockHttpResponse(request, 200, "abc"));
```

Comment:
nit, personally I prefer the `bodyBytes` override, as it be clear about what is the exact response payload.

Reply:
Seems azure-json can't handle the case where body is just a number.. It has to be a string at least.
Updated.

Reply:
The body must be bytes of UTF8 encoded string.

So, 1st is correct `"0".toBytes()`. 2nd should be `"null".toBytes()`. 3nd be `"\"abc\"".toBytes()`.

There is even `0.toBytes()`

Reply:
fixed